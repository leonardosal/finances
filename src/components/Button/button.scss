.button {
  background-color: #6200ee;
  color: #fff;
  border-radius: 24px;
  padding: 16px 22px;
  font-size: 0.8rem;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.4);
  outline: none;

  &:active,
  &:hover,
  &:focus {
    background-color: #3f098b;
  }
}
