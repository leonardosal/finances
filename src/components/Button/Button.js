import React from 'react';
import propTypes from 'prop-types';

import './button.scss';

const Button = ({ id, label, type, onClick }) => (
  <button id={id} type={type} className="button" onClick={onClick}>
    {label}
  </button>
);

Button.propTypes = {
  id: propTypes.string,
  label: propTypes.string,
  onClick: propTypes.func,
  type: propTypes.oneOf(['button', 'submit', 'reset']),
};

Button.defaultProps = {
  id: '',
  label: '',
  type: 'button',
  onClick: () => null,
};

export default Button;
