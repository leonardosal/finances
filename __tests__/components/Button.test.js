import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../../src/components/Button';

it('should render button component with label "Add"', () => {
  const props = {
    label: 'Add',
    onClick: () => true,
  };
  const component = mount(<Button label={props.label} />);
  expect(toJson(component)).toMatchSnapshot();
});
