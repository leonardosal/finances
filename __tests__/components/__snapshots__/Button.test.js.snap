// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button component with label "Add" 1`] = `
<Button
  label="Add"
  onClick={[Function]}
  type="button"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</Button>
`;
